servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletBuilder.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    /**
+     * Creates a new instance.
+     */
+    public ServletBuilder(ServerBuilder serverBuilder, String contextPath) {
+        requireNonNull(serverBuilder, "serverBuilder");
+        requireNonNull(contextPath, "contextPath");
+        if (!contextPath.isEmpty()) {
+            checkArgument(contextPath.charAt(0) == '/' && contextPath.charAt(contextPath.length() - 1) != '/',
+                          "contextPath must start with / and must not end with /. contextPath: %s",
+                          contextPath);
+        }
+        servletContext = new DefaultServletContext(contextPath);
+        this.contextPath = contextPath;
+        this.serverBuilder = serverBuilder;
+    }
+
+    /**
+     * Creates a new {@link ServerBuilder}.
+     */
+    public ServerBuilder build() {
+        final String path = contextPath.isEmpty() ? "/" : contextPath;
+        final DefaultServletService servletService = new DefaultServletService(servletContext);
+        serverBuilder.serviceUnder(path, servletService);
+        if (rootServletAdded) {
+            serverBuilder.service(path, servletService);
+        } else {
+            serverBuilder.service(path, (ctx, req) -> HttpResponse.of(HttpStatus.NOT_FOUND));
+        }
+        servletContext.init();
+        return serverBuilder;
+    }
+
+    /**
+     * Add a servlet.
+     */
+    public ServletBuilder servlet(String path, HttpServlet httpServlet) {
+        requireNonNull(path, "path");
+        requireNonNull(httpServlet, "httpServlet");
+        validatePath(path);
+        servletContext.addServlet(contextPath + path, httpServlet);
+        return this;
+    }
+
+    /**
+     * Add a servlet.
+     */
+    public ServletBuilder servlet(String path, String servletClass) {
+        requireNonNull(path, "path");
+        requireNonNull(servletClass, "servletClass");
+        validatePath(path);
+        servletContext.addServlet(contextPath + path, servletClass);
+        return this;
+    }
+
+    /**
+     * Validate servlet path.
+     */
+    public void validatePath(String path) {
+        if (path.isEmpty() || "/".equals(path)) {
+            rootServletAdded = true;
+        } else {
+            checkArgument(path.charAt(0) == '/',
+                          "servletPath must start with /. servletPath: %s",
+                          path);
+        }
+    }
+
+    /**
+     * Set attribute value.
+     */
+    public ServletBuilder attribute(String key, @Nullable Object value) {
+        requireNonNull(key, "key");
+        servletContext.setAttribute(key, value);
+        return this;
+    }
+
+    /**
+     * Set init parameter.
+     */
+    public ServletBuilder initParameter(String key, @Nullable String value) {
+        requireNonNull(key, "key");
+        servletContext.setInitParameter(key, value);
+        return this;
+    }
+
+    /**
+     * Add a new mime mapping.
+     */
+    public ServletBuilder mimeMapping(MimeMappings mimeMappings) {
```

Comment:
We should not expose this setter because `MimeMappings` is a package-private class.
How about adding a method for adding mime mapping in this class:
```java
public void addMimeMapping(String extension, String mimeType) {
    ...
}
```

Reply:
But I think MimeMapping usually add a large extensions list (thousand of extensions). Therefore I want to separate another methods to add mime mapping instead of use ServletBuilder.

Reply:
Then we can add
```java
addMimeMappings(Map<String, String> mappings) {...}
```
What do you think: